services/userservice/userservice.go

```diff
@@ -0,0 +1,46 @@
+package userservice
```

Comment:
not a huge fan of service, but can't think of anything better. Anything else you were considering?

Reply:
i'm honestly not sure if i like the way the codebase is structured. In the next branch I also called things handlers instead of controllers (as the MVC pattern says we should do).

I'm open to suggestions. I'm not a huge fan of `userutil`. Just `user` might be okay, but it's in use by the model package.

Reply:
what if we just put all these functions into the user package? 
or potentially just put it into the `usercontroller` file, does anything other than the user controller need these functions? 

Other not great list of names
`userwrapper`, `usermanager`, `userhelper`